gh-57684: Add -P cmdline option and PYTHONSAFEPATH env var (#31542)
python/cpython: Doc/using/cmdline.rst

      Modify ``.pyc`` filenames according to :pep:`488`.

.. cmdoption:: -P

>    Don't add :data:`sys.path[0] <sys.path>`: don't prepend the current working
>    directory (``python -m module``), an empty string (``python -c code``) or
>    the first command line argument (``python script.py``) to :data:`sys.path`.

Reviewer: ```suggestion
   Do not prepend the initial working directory (``python -m module``), the
   current working directory (``python -c code``), or the directory of the
   executed script (``python script.py``) to :data:`sys.path`.
```
I think saying "prepend" suffices instead of talking about adding `sys.path[0]`. 

With the `-m` option, `sys.path[0]` is the full path of the initial working directory. With the `-c` option and the REPL, the current working directory is prepended to `sys.path`. It's an implementation detail that it's represented in this case as an empty string. It's not exactly an obscure implementation detail, since most people know that `normpath('')` -> `'.'`, but I think it's better to be explicit about the behavior.

      Modify ``.pyc`` filenames according to :pep:`488`.

.. cmdoption:: -P

   Don't prepend a potentially unsafe path to :data:`sys.path`:

   * ``python -m module`` command line: Don't prepend the current working
     directory.

>    * ``python script.py`` command line: Don't prepend the script path.

Reviewer 2: ```suggestion
   * ``python script.py`` command line: Don't prepend the script's directory.
```

   within a Python program as the variable :data:`sys.path`.

> .. envvar:: PYTHONDONTADDPATH0

Reviewer: You're emphasizing `sys.path[0]` again. There's always a `sys.path[0]`. How about `PYTHONUSESAFEPATH`?

Reviewer 3: How does the doc call the value that is prepended?  Main script base directory?  Could be  `PYTHONDONTADDMAINDIR` or `PYTHONNOMAINDIR`.

(I wondered about including `IMPORT` or `PATH` but then you get something clunky or you add prepositions and get too long)
